ArcGISRuntimeSDKQt_CppSamples/Layers/ListKmlContents/ListKmlContents.cpp

```diff
@@ -313,22 +288,26 @@ void ListKmlContents::setSceneView(SceneQuickView* sceneView)
   emit sceneViewChanged();
 }
 
-QStringList ListKmlContents::levelNodeNames() const
+QStringList ListKmlContents::levelNodeNames()
 {
   return m_levelNodeNames;
 }
 
-QString ListKmlContents::labelText() const
+QString ListKmlContents::labelText()
 {
+  m_labelText.clear();
```

Comment:
You don't need to keep `m_labelText` to do this recurisve function.

Try:


```cpp
// recursively build string to indicate node's ancestors
QStringList ListKmlContents::buildPathLabel(KmlNode* node) const
{
  if (node)
    return buildPathLabel(node->parentNode()) << node->name();
  else
    return QStringList { };
}
```

With:

```cpp
QString ListKmlContents::labelText() const 
{
  return buildPathLabel(m_currentNode).join(" > ");
}
```